developer_tools/validate_xml.py

```diff
@@ -29,29 +30,33 @@ def validateTests():
   """
   print 'Beginning test validation...'
   tests = get_coverage_tests.getRegressionTests(skipExpectedFails=True)
-  res=[0,0,0] #run, pass, fail
-  failed={}
+  res = [0, 0, 0] #run, pass, fail
+  failed = {}
   devnull = open(os.devnull, "wb")
-  for dir,files in tests.items():
+  for dir, files in tests.items():
     #print directory being checked'
-    checkmsg = ' Directory: '+dir
-    print colors.neutral+checkmsg.rjust(maxlen,'-')
+    checkmsg = ' Directory: ' + dir
+    print colors.neutral + checkmsg.rjust(maxlen, '-')
     #check files in directory
     for f in files:
-      fullpath = os.path.join(dir,f)
-      res[0]+=1
-      startmsg =  'Validating '+f
+      fullpath = os.path.join(dir, f)
+      # check if input exists; if not, it probably gets created by something else, and can't get checked here
```

Comment:
?

Reply:
The input file to be run is created by the template, so it can't be validated when "validate_xml.py" is run. This caused a total crash previously; this modification avoids such a crash.

What if a required test file is skipped in the validation but doesn't get created later when another test is run? Well, that's not the validation's job to check, that's the test's job to check, so I think skipping missing test files here should be acceptable.

Reply:
Another useful future feature would be to run the validation script as part of the test harness before each test is run, instead of as a separate procedure.

Reply:
We don't find files to validate by what exists; we find files to validate by what is requested in the "tests" files.

Reply:
Ohhh...right...sorry